src/main/java/io/strimzi/kafka/bridge/http/HttpSourceBridgeEndpoint.java

```diff
@@ -95,8 +96,8 @@ public void handle(Endpoint<?> endpoint) {
 
             for (int i = 0; i < sendHandlers.size(); i++) {
                 // check if, for each future, the sending operation is completed successfully or failed
-                if (done.result() != null && done.result().succeeded(i)) {
-                    RecordMetadata metadata = done.result().resultAt(i);
+                if (((CompositeFutureImpl) done).resultAt(i) != null) {
+                    RecordMetadata metadata = ((CompositeFutureImpl) done).resultAt(i);
```

Comment:
I don't think we need to do this hack casting to the actual `CompositeFutureImpl` class.
In this specific use case where an overall operation can have mixed failures and success, we have to use the `sendHandlers` list directly so the following code will work ...

```java
for (int i = 0; i < sendHandlers.size(); i++) {
                // check if, for each future, the sending operation is completed successfully or failed
                if (sendHandlers.get(i).succeeded() && sendHandlers.get(i).result() != null) {
                    RecordMetadata metadata = (RecordMetadata) sendHandlers.get(i).result();
                    log.debug("Delivered record {} to Kafka on topic {} at partition {} [{}]", finalRecords.get(i), metadata.getTopic(), metadata.getPartition(), metadata.getOffset());
                    results.add(new HttpBridgeResult<>(metadata));
                } else {
                    String msg = sendHandlers.get(i).cause().getMessage();
                    int code = getCodeFromMsg(msg);
                    log.error("Failed to deliver record " + finalRecords.get(i) + " due to {}", sendHandlers.get(i).cause());
                    results.add(new HttpBridgeResult<>(new HttpBridgeError(code, msg)));
                }
            }
```